zenoh-flow/src/io/input.rs

```diff
@@ -195,23 +200,38 @@ impl InputRaw {
     }
 }
 
-#[derive(Clone, Debug)]
-pub struct Input<T: ZFData + 'static> {
-    _phantom: PhantomData<T>,
+/// This function is what Zenoh-Flow will use to deserialize the data received on the `Input`.
+///
+/// It will be called for instance when data is received serialized (i.e. from an upstream node that
+/// is either not implemented in Rust or on a different process) before it is given to the user's
+/// code.
+pub(crate) type DeserializerFn<T> = dyn Fn(&[u8]) -> anyhow::Result<T> + Send + Sync;
```

Comment:
This should be defined in the same place we define `SerializerFn`, so in `zenoh-flow/src/types/message.rs`